python/amici/petab_objective.py

```diff
@@ -352,14 +352,24 @@ def create_parameter_mapping_for_condition(
         condition_map_sim[PREEQ_INDICATOR_ID] = 0.0
         condition_scale_map_sim[PREEQ_INDICATOR_ID] = LIN
 
-        def _set_initial_concentration(condition_id, species_id, init_par_id,
-                                       par_map, scale_map):
+        def _set_initial_state(condition_id, element_id, init_par_id,
+                               par_map, scale_map):
             value = petab.to_float_if_float(
-                petab_problem.condition_df.loc[condition_id, species_id])
+                petab_problem.condition_df.loc[condition_id, element_id])
             if pd.isna(value):
-                value = get_species_initial(
-                    petab_problem.sbml_model.getSpecies(species_id)
-                )
+                element = petab_problem.sbml_model.getElementBySId(element_id)
```

Comment:
hmm this doesn't account for InitialAssignments right? (was already the case for species before, but I am not sure thats correct in the first place)

Reply:
Yes, you are right. Good catch. Should add a test case.

Reply:
Fixed